torch/_inductor/scheduler.py

```diff
@@ -1522,9 +1528,16 @@ def remove_kernel_local_buffers(self):
         same kernel can be removed.
         """
 
-        names_to_remove = (
-            V.kernel.store_buffer_names & self.buffer_names_no_longer_needed
-        )
+        fused_node_names = {
+            node.get_name()
+            for node in V.kernel.node_schedule
```

Comment:
Actually I think in most cases `fused_node_names` should be equivalent to `V.kernel.store_buffer_names`. But I'm not sure if it's always true for even foreach kernel/cpp kernel etc.

Reply:
Actually yeah that sounds right to me. I think in the case of `ForeachKernel` this function is only called for the sub-kernel, which is just a `TritonKernel`.